### ISSUE TITLE
Add support for renaming volume names

### ISSUE THREAD

**Issue by prashanthpai** [gluster/glusterd2#310]
Use volume id internally which should allow for supporting renaming of volume names.

**Comment by atinmu** [comment]
## Checklist
- [ ] ReST APIs availibility in wiki
- [ ] CLI commands
- [ ] Sufficient logging
- [ ] Availibility of metrics data if applicable
- [ ] Xlator options table upto date?
- [ ] Test cases & test coverage of the functionality

**Comment by aravindavk** [comment]
Closing this issue since no use case for changing volume name. Please re-open the issue if this needs to be supported.

**Comment by rs-fabrica** [comment]
> Closing this issue since no use case for changing volume name. Please re-open the issue if this needs to be supported.

Use case:

I created a Gluster volume to hold the Docker volumes for a Docker Swarm Stack. I had to rename the stack so I would like to rename the Gluster volume also.

Some context: the stacks that we decided deserve exclusive Gluster volumes should have Gluster volumes with the same name as the stacks they support by internal policy. Not all stacks that have their volumes in Gluster are using exclusive Gluster volumes but for the ones that are, we would like to keep stack name = gluster volume name.

Obviously I can create a new Glsuter volume with the new name and redo the +4h copy process but a simialr solution can probably be used for every rename operation. The point of a rename operation is to save this unnecessary work.
